guides/hyperlane-use-tia-for-gas.md

```diff
@@ -0,0 +1,437 @@
+
+```bash
+wasmd keys add my-key
+```
+
+To get your Stride address, import the generated seed phrase to the `strided` CLI using:
+
+```bash
+strided keys add --recover my-key
+```
+
+To get the private key, run:
+
+```bash
+wasmd keys export --unsafe --unarmored-hex my-key
+```
+
+:::
+
+### Deploy local data availability service and rollup
+
+<!-- TODO: Update to a forked version that has this new image -->
+
+Create a Dockerfile in the `wasmd` repo that will be used to run the rollup.
+
+```bash
+wget -O Dockerfile https://raw.githubusercontent.com/Stride-Labs/Rollkit-docs/hyperlane-v2/guides/assets/hyperlane-use-tia-for-gas/wasmd.Dockerfile
+```
+
+Fork the cw-hyperlane repo
+
+```bash
+# Back out to the same level as the wasmd repo
+cd ..
+
+git clone --depth 1 git@github.com:many-things/cw-hyperlane.git
+cd cw-hyperlane
+git checkout 4f5656d4704178ac54d10467ca7edc3df2312c4b
+```
+
+Create the docker compose file that will be used for the data availability service, the localwasm chain, and the Hyperlane validators and relayers.
+
+```bash
+wget -O example/docker-compose.yml https://raw.githubusercontent.com/Stride-Labs/Rollkit-docs/hyperlane-v2/guides/assets/hyperlane-use-tia-for-gas/docker-compose.yml
+```
+
+To start the data availability service, run:
+
+```bash
+docker compose -f example/docker-compose.yml up da
+```
+
+Then start the localwasm chain:
+
+```bash
+docker compose -f example/docker-compose.yml up localwasm
+```
+
+With that, we have kickstarted our `wasmd` network! 🎉
+
+### Fund our accounts
+
+#### Stride Account
+
+If you're using the account specified in this guide, the stride account should already be funded!
+
+If you're using a new account, you can fund it from the faucet at [Stride testnet faucet](https://stride-faucet.pages.dev).
+
+<!-- TODO: Add TIA on stride to faucet -->
+
+#### Rollup Account
+
+We can fund the rollup account through one of the genesis accounts in the docker container
+
+```bash
+docker exec -it localwasm \
+  wasmd tx bank send localwasm-key wasm133xh839fjn9wxzg6vhc0370lcem8939zr8uu45 \
+  10000000uwasm -y --gas auto --gas-adjustment 1.5 --gas-prices 0.025uwasm \
+  --keyring-backend test --node http://localhost:36657
+```
+
+:::info
+`localwasm-key` is the genesis validator's address. See https://rollkit.dev/cosmwasm/init.sh for more info.
+:::
+
+:::tip
+To check if the transaction was successful, get the `txhash` field from the response to the `wasmd tx bank send` command above, and run:
+
+```bash
+wasmd q tx $TXHASH
+```
+
+The transaction was successful if the `code` field is 0 (success).
+:::
+
+## Deploy Hyperlane
+
+### Deploy Contracts
+
+#### Create `config.yaml` with our networks setup:
+
+<!-- TODO: Update the fee to be denominated in the tokenfactory TIA -->
+
+```bash
+wget -O config.yaml https://raw.githubusercontent.com/Stride-Labs/Rollkit-docs/hyperlane-v2/guides/assets/hyperlane-use-tia-for-gas/hyperlane-config.yaml
```

Comment:
For all these `wget` commands I think we should just inline the content and have the user copy into a local file. 

This eliminates the need for all the asset files that are part of this PR. 

Also these `wget` commands are problematic in that they aren't targeting the actual files being upload but some branch on the Strife-Labs fork. A little chicken and egg situation. 

This is going to lead to the tutorial potentially breaking without warning.